stats.go

```diff
@@ -112,3 +114,25 @@ func (s *Stats) Snapshot() *Stats {
 	})
 	return snap
 }
+
+func (s *Stats) String() string {
```

Comment:
I needed this for a debug error statement, not json serialization.  Nowhere do we serialize any of this as json so seemed unnecessary at this point.

It could be refactored to be marshal/unmarshal, and the `String` method could then use unmarshal it.

Fundamentally though, this is only for debugging, I'm hoping it goes away once we figure out the bug.  It was not intended for long term use.